src/audio/template_comp/template-generic.c

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2025 Intel Corporation.
+
+#include <sof/audio/module_adapter/module/generic.h>
+#include <sof/audio/component.h>
+#include <sof/audio/sink_api.h>
+#include <sof/audio/sink_source_utils.h>
+#include <sof/audio/source_api.h>
+#include <stdint.h>
+#include "template.h"
+
+#if CONFIG_FORMAT_S16LE
+/**
+ * template_comp_s16() - Process S16_LE format.
+ * @mod: Pointer to module data.
+ * @source: Source for PCM samples data.
+ * @sink: Sink for PCM samples data.
+ * @frames: Number of audio data frames to process.
+ *
+ * This is the processing function for 16-bit signed integer PCM formats. The
+ * audio samples in every frame are re-order to channels order defined in
+ * component data channels_order[].
+ *
+ * Return: Value zero for success, otherwise an error code.
+ */
+static int template_comp_s16(const struct processing_module *mod,
+			     struct sof_source *source,
+			     struct sof_sink *sink,
+			     uint32_t frames)
+{
+	struct template_comp_comp_data *cd = module_get_private_data(mod);
+	int16_t *x, *x_start, *x_end;
+	int16_t *y, *y_start, *y_end;
+	size_t size;
+	int x_size, y_size;
+	int source_samples_without_wrap;
+	int samples_without_wrap;
+	int samples = frames * cd->channels;
+	int bytes = frames * cd->frame_bytes;
+	int ret;
+	int ch;
+	int i;
+
+	/* Get pointer to source data in circular buffer, get buffer start and size to
+	 * check for wrap. The size in bytes is converted to number of s16 samples to
+	 * control the samples process loop. If the number of bytes requested is not
+	 * possible, an error is returned.
+	 */
+	ret = source_get_data(source, bytes, (void const **)&x, (void const **)&x_start, &size);
+	x_size = size >> 1; /* Bytes to number of s16 samples */
+	if (ret)
+		return ret;
+
+	/* Similarly get pointer to sink data in circular buffer, buffer start and size. */
+	ret = sink_get_buffer(sink, bytes, (void **)&y, (void **)&y_start, &size);
+	y_size = size >> 1; /* Bytes to number of s16 samples */
+	if (ret)
+		return ret;
```

Comment:
ditto, and in other formats below too of course

Reply:
Yep, fixed now.